From 89a976959e6a84b363ea2c5e53685b2e2b810711 Mon Sep 17 00:00:00 2001
From: Maarten de Vries <maarten@de-vri.es>
Date: Fri, 11 May 2018 07:55:17 +0200
Subject: [PATCH] Fix allocator type to pass GCC8 static assert. (#888)

Upstream-Status: Backport [from kinetic, https://github.com/ros-planning/moveit/commit/89a976959e6a84b363ea2c5e53685b2e2b810711]

This patch has been generated with:
  `git format-patch -1 89a976959e6a84b363ea2c5e53685b2e2b810711`
in the moveit repository.

Signed-off-by: Lukas Bulwahn <lukas.bulwahn@gmail.com>
---
 moveit_core/robot_model/include/moveit/robot_model/link_model.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/moveit_core/robot_model/include/moveit/robot_model/link_model.h b/moveit_core/robot_model/include/moveit/robot_model/link_model.h
index d8aec2f..3f2b1b6 100644
--- a/moveit_core/robot_model/include/moveit/robot_model/link_model.h
+++ b/moveit_core/robot_model/include/moveit/robot_model/link_model.h
@@ -65,7 +65,7 @@ typedef std::map<std::string, const LinkModel*> LinkModelMapConst;
 
 /** \brief Map from link model instances to Eigen transforms */
 typedef std::map<const LinkModel*, Eigen::Affine3d, std::less<const LinkModel*>,
-                 Eigen::aligned_allocator<std::pair<const LinkModel*, Eigen::Affine3d> > >
+                 Eigen::aligned_allocator<std::pair<const LinkModel* const, Eigen::Affine3d> > >
     LinkTransformMap;
 
 /** \brief A link from the robot. Contains the constant transform applied to the link and its geometry */
-- 
2.7.4

